src/index.ts

```diff
@@ -2,25 +2,13 @@
 // SPDX-License-Identifier: Apache-2.0
 
 
-import * as child_process from 'child_process';
 import * as path from 'path';
-import { aws_ec2 as ec2, aws_iam as iam, aws_lambda as lambda, Duration, CustomResource, Token } from 'aws-cdk-lib';
-import { PolicyStatement, AddToPrincipalPolicyResult } from 'aws-cdk-lib/aws-iam';
+import { custom_resources as cr, aws_ec2 as ec2, aws_iam as iam, aws_lambda as lambda, Duration, CustomResource, Token, Stack } from 'aws-cdk-lib';
+import { AddToPrincipalPolicyResult, PolicyStatement } from 'aws-cdk-lib/aws-iam';
 import { Construct } from 'constructs';
-import { shouldUsePrebuiltLambda } from './config';
 
 export interface ECRDeploymentProps {
 
-  /**
-   * Image to use to build Golang lambda for custom resource, if download fails or is not wanted.
-   *
-   * Might be needed for local build if all images need to come from own registry.
-   *
-   * Note that image should use yum as a package manager and have golang available.
-   *
-   * @default public.ecr.aws/sam/build-go1.x:latest
-   */
-  readonly buildImage?: string;
```

Comment:
For maximum user impact, it would be better to leave this parameter in place, and to not bump the Major Version. That way, all users currently using the affected version can transparently upgrade to a fixed version. Ignore the property if supplied and report a warning.

Then, in a future Major Version you can remove this parameter, to have a correct API contract.